### PR TITLE
[ci] Re-enable Windows test on premerge

### PR DESCRIPTION
**Pull request by khluu**
## Why are these changes needed?
- These tests were added a condition to only run when there's the continuous build flag on, which also disables it on premerge.

## Changes
- I have added an extra condition to have these tests run if the buildkite pipeline is `premerge`. 

## Checks
- This PR premerge check shows that the Windows tests are showing up.

### CODE REVIEW

File: .buildkite/core.rayci.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ steps:
         --except-tags debug_tests,asan_tests,post_wheel_build,ha_integration,mem_pressure,tmpfs,container,manual
 
   - label: ":ray: core: :windows: python tests"
-    if: build.env("RAYCI_CONTINUOUS_BUILD") == "1" || build.pull_request.labels includes "continuous-build"
+    if: build.env("BUILDKITE_PIPELINE_ID") == "0189942e-0876-4b8f-80a4-617f988ec59b" || build.env("RAYCI_CONTINUOUS_BUILD") == "1" || build.pull_request.labels includes "continuous-build"
```
Comment (aslonnie) [inline .buildkite/core.rayci.yml:74]: remove `|| build.pull_request.labels includes "continuous-build"` ? if it is a PR, it now runs on premerge and will always run?

**Review by aslonnie — APPROVED** [review]

File: .buildkite/core.rayci.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ steps:
         --except-tags debug_tests,asan_tests,post_wheel_build,ha_integration,mem_pressure,tmpfs,container,manual
 
   - label: ":ray: core: :windows: python tests"
-    if: build.env("BUILDKITE_PIPELINE_ID") == "0189942e-0876-4b8f-80a4-617f988ec59b" || build.env("RAYCI_CONTINUOUS_BUILD") == "1" || build.pull_request.labels includes "continuous-build"
+    if: build.env("BUILDKITE_PIPELINE_ID") == "0189942e-0876-4b8f-80a4-617f988ec59b" || build.env("RAYCI_CONTINUOUS_BUILD") == "1"
```
Comment (aslonnie) [inline .buildkite/core.rayci.yml:74]: rather than using `build.env("BUILDKITE_PIPELINE_ID")`, just use `pipeline.id` ?

Comment (aslonnie) [reply .buildkite/core.rayci.yml:74]: see https://buildkite.com/docs/pipelines/conditionals#variable-and-syntax-reference-variables for supported builtin variables.

Comment (khluu) [reply .buildkite/core.rayci.yml]: Thank you! I've updated them all to use `pipeline.id`. Will merge the PR after the tests run.

**Pull request merged**